### ISSUE TITLE
kernel BUG at fs/bcachefs/ec.c:437

### ISSUE THREAD

**Issue by RAOF** [koverstreet/bcachefs#177]
When attempting to perform a `dpkg --install` operation on my system I reliably get the following kernel bug:

```
[10608.079874] ------------[ cut here ]------------
[10608.079934] kernel BUG at fs/bcachefs/ec.c:437!
[10608.079955] invalid opcode: 0000 [#1] SMP PTI
[10608.079964] CPU: 1 PID: 9713 Comm: kworker/u16:15 Tainted: P           O      5.10.0+bcachefs.git20201231.31e3b794-1-generic #1-Ubuntu
[10608.079970] Hardware name: System manufacturer System Product Name/Z170 PRO GAMING, BIOS 1904 07/05/2016
[10608.080050] Workqueue: events_unbound __bch2_read_endio [bcachefs]
[10608.080141] RIP: 0010:bch2_ec_read_extent+0x514/0x550 [bcachefs]
[10608.080151] Code: 85 c0 0f 85 41 f9 04 00 48 8b 85 b0 fe ff ff f0 4c 0f b3 38 e9 17 ff ff ff 0f 0b 85 c0 0f 84 03 fc ff ff e9 04 fc ff ff 0f 0b <0f> 0b 0f 0b 48 c7 c6 00 40 b8 c1 48 c7 c7 e0 51 b9 c1 e8 a5 5e 0d
[10608.080156] RSP: 0018:ffffbe6fc2027888 EFLAGS: 00010246
[10608.080164] RAX: ffff9ff3d68b20d0 RBX: ffff9ff2c2991f80 RCX: ffff9ff39f2b8000
[10608.080171] RDX: 0000000000000002 RSI: ffffffffc1b10abd RDI: 0000000000000002
[10608.080177] RBP: ffffbe6fc20279e0 R08: 00000fffffffffff R09: 0000000000000003
[10608.080182] R10: 0000000000000000 R11: 0000000000000100 R12: ffff9ff224e00000
[10608.080189] R13: ffff9ff3d68b2000 R14: ffff9ff3d68b20c0 R15: 0000000000000000
[10608.080196] FS:  0000000000000000(0000) GS:ffff9ff7e6c40000(0000) knlGS:0000000000000000
[10608.080202] CS:  0010 DS: 0000 ES: 0000 CR0: 0000000080050033
[10608.080208] CR2: 000055f062120ca8 CR3: 000000039e410001 CR4: 00000000003706e0
[10608.080214] DR0: 0000000000000000 DR1: 0000000000000000 DR2: 0000000000000000
[10608.080219] DR3: 0000000000000000 DR6: 00000000fffe0ff0 DR7: 0000000000000400
[10608.080223] Call Trace:
[10608.080241]  ? __kmalloc_track_caller+0x114/0x240
[10608.080317]  ? bch2_ec_read_extent+0x93/0x550 [bcachefs]
[10608.080380]  __bch2_read_extent+0xa00/0x10c0 [bcachefs]
[10608.080438]  bch2_read_retry+0x2a4/0x590 [bcachefs]
[10608.080494]  ? bch2_read_retry+0x9c/0x590 [bcachefs]
[10608.080508]  ? kfree+0x3b2/0x3e0
[10608.080561]  bch2_rbio_retry+0x125/0x200 [bcachefs]
[10608.080614]  __bch2_read_endio+0x47a/0x670 [bcachefs]
[10608.080624]  ? __switch_to+0x11d/0x3f0
[10608.080635]  ? __switch_to_asm+0x36/0x70
[10608.080645]  process_one_work+0x1ef/0x390
[10608.080655]  worker_thread+0x50/0x370
[10608.080662]  kthread+0x12f/0x150
[10608.080668]  ? process_one_work+0x390/0x390
[10608.080674]  ? __kthread_bind_mask+0x70/0x70
[10608.080683]  ret_from_fork+0x22/0x30
[10608.080689] Modules linked in: poly1305_generic libpoly1305 poly1305_x86_64 chacha_generic chacha_x86_64 libchacha bcachefs crc64 xor raid6_pq libcrc32c nfnetlink zfs(PO) zunicode(PO) zavl(PO) icp(PO) zcommon(PO) znvpair(PO) spl(O) zlua(PO) intel_rapl_msr intel_rapl_common x86_pkg_temp_thermal intel_powerclamp coretemp kvm_intel snd_hda_codec_realtek kvm snd_hda_codec_generic ledtrig_audio snd_hda_codec_hdmi snd_hda_intel snd_intel_dspcfg soundwire_intel soundwire_generic_allocation soundwire_cadence snd_hda_codec snd_hda_core snd_hwdep soundwire_bus snd_soc_core mei_hdcp snd_compress ac97_bus snd_pcm_dmaengine snd_pcm joydev hid_multitouch snd_seq_midi snd_seq_midi_event snd_rawmidi snd_seq rapl snd_seq_device input_leds intel_cstate snd_timer snd soundcore eeepc_wmi asus_wmi sparse_keymap serio_raw wmi_bmof efi_pstore ee1004 mei_me mei acpi_pad mac_hid sch_fq_codel msr parport_pc ppdev lp parport ip_tables x_tables autofs4 overlay nls_utf8 isofs nls_iso8859_1 dm_mirror dm_region_hash
[10608.080849]  dm_log amdgpu iommu_v2 gpu_sched hid_logitech_hidpp hid_logitech_dj hid_generic usbhid hid uas usb_storage radeon nouveau mxm_wmi i2c_algo_bit ttm drm_kms_helper crct10dif_pclmul crc32_pclmul syscopyarea ghash_clmulni_intel sysfillrect aesni_intel sysimgblt fb_sys_fops cec rc_core crypto_simd cryptd glue_helper nvme e1000e psmouse ahci drm libahci i2c_i801 nvme_core xhci_pci i2c_smbus xhci_pci_renesas wmi video
[10608.080938] ---[ end trace 0f3bb942023ee4d5 ]---
[10609.089137] RIP: 0010:bch2_ec_read_extent+0x514/0x550 [bcachefs]
[10609.089139] Code: 85 c0 0f 85 41 f9 04 00 48 8b 85 b0 fe ff ff f0 4c 0f b3 38 e9 17 ff ff ff 0f 0b 85 c0 0f 84 03 fc ff ff e9 04 fc ff ff 0f 0b <0f> 0b 0f 0b 48 c7 c6 00 40 b8 c1 48 c7 c7 e0 51 b9 c1 e8 a5 5e 0d
[10609.089140] RSP: 0018:ffffbe6fc2027888 EFLAGS: 00010246
[10609.089141] RAX: ffff9ff3d68b20d0 RBX: ffff9ff2c2991f80 RCX: ffff9ff39f2b8000
[10609.089142] RDX: 0000000000000002 RSI: ffffffffc1b10abd RDI: 0000000000000002
[10609.089143] RBP: ffffbe6fc20279e0 R08: 00000fffffffffff R09: 0000000000000003
[10609.089143] R10: 0000000000000000 R11: 0000000000000100 R12: ffff9ff224e00000
[10609.089144] R13: ffff9ff3d68b2000 R14: ffff9ff3d68b20c0 R15: 0000000000000000
[10609.089145] FS:  0000000000000000(0000) GS:ffff9ff7e6c40000(0000) knlGS:0000000000000000
[10609.089146] CS:  0010 DS: 0000 ES: 0000 CR0: 0000000080050033
[10609.089147] CR2: 000055f062120ca8 CR3: 0000000241a38005 CR4: 00000000003706e0
[10609.089148] DR0: 0000000000000000 DR1: 0000000000000000 DR2: 0000000000000000
[10609.089148] DR3: 0000000000000000 DR6: 00000000fffe0ff0 DR7: 0000000000000400
```

The relevant part of the backtrace is:
```
__kmalloc_track_caller+0x114/0x240:
slab_alloc_node at mm/slub.c:2893
(inlined by) slab_alloc at mm/slub.c:2899
(inlined by) __kmalloc_track_caller at mm/slub.c:4464

bch2_ec_read_extent+0x93/0x550:
bch2_ec_read_extent at /build/linux-YWSD8U/linux-5.10.0+bcachefs.git20201231.31e3b794/fs/bcachefs/ec.c:418

__bch2_read_extent+0xa00/0x10c0:
__bch2_read_extent at /build/linux-YWSD8U/linux-5.10.0+bcachefs.git20201231.31e3b794/fs/bcachefs/io.c:2241

bch2_read_retry+0x2a4/0x590:
bch2_read_retry at /build/linux-YWSD8U/linux-5.10.0+bcachefs.git20201231.31e3b794/fs/bcachefs/io.c:1711

bch2_read_retry+0x9c/0x590:
bkey_on_stack_realloc at /build/linux-YWSD8U/linux-5.10.0+bcachefs.git20201231.31e3b794/fs/bcachefs/bkey_on_stack.h:17
(inlined by) bkey_on_stack_reassemble at /build/linux-YWSD8U/linux-5.10.0+bcachefs.git20201231.31e3b794/fs/bcachefs/bkey_on_stack.h:26
(inlined by) bch2_read_retry at /build/linux-YWSD8U/linux-5.10.0+bcachefs.git20201231.31e3b794/fs/bcachefs/io.c:1689
```

**Comment by RAOF** [comment]
I always have the upstream commit this was built from in my kernel version, so you can read it out of dmesg; for clarity's sake, this is on commit `31e3b794`

**Comment by koverstreet** [comment]
This code path was triggered by the pointer we're doing a recovery read from not matching the stripe the stripe pointer pointed to. It's been changed to not trigger a BUG_ON(), along with a whole bunch of other stuff.

Let me know what the current master branch does for you.

**Comment by RAOF** [comment]
The current branch hits a `BUG_ON()` earlier, during mount:

```
[  735.788602] bcachefs (ea4e6e4c-5851-40e2-99f6-31b4f34e1feb): invalid journal entry: last_seq > seq, fixing
[  735.788615] bcachefs (ea4e6e4c-5851-40e2-99f6-31b4f34e1feb): invalid journal entry: last_seq > seq, fixing
[  735.788836] bcachefs (ea4e6e4c-5851-40e2-99f6-31b4f34e1feb): invalid journal entry: last_seq > seq, fixing
[  735.790215] bcachefs (ea4e6e4c-5851-40e2-99f6-31b4f34e1feb): invalid journal entry: last_seq > seq, fixing
[  735.790261] bcachefs (ea4e6e4c-5851-40e2-99f6-31b4f34e1feb): invalid journal entry: last_seq > seq, fixing
[  735.790279] bcachefs (ea4e6e4c-5851-40e2-99f6-31b4f34e1feb): invalid journal entry: last_seq > seq, fixing
[  735.790295] bcachefs (ea4e6e4c-5851-40e2-99f6-31b4f34e1feb): invalid journal entry: last_seq > seq, fixing
[  735.790313] bcachefs (ea4e6e4c-5851-40e2-99f6-31b4f34e1feb): invalid journal entry: last_seq > seq, fixing
[  735.790329] bcachefs (ea4e6e4c-5851-40e2-99f6-31b4f34e1feb): invalid journal entry: last_seq > seq, fixing
[  735.790463] bcachefs (ea4e6e4c-5851-40e2-99f6-31b4f34e1feb): invalid journal entry: last_seq > seq, fixing
[  735.790465] bcachefs (ea4e6e4c-5851-40e2-99f6-31b4f34e1feb): Ratelimiting new instances of previous error
[  744.505730] bcachefs (ea4e6e4c-5851-40e2-99f6-31b4f34e1feb): journal read done, 553154 keys in 130 entries, seq 6537004
[  744.871532] bcachefs (ea4e6e4c-5851-40e2-99f6-31b4f34e1feb): starting alloc read
[  746.154936] bcachefs (ea4e6e4c-5851-40e2-99f6-31b4f34e1feb): alloc read done
[  746.154937] bcachefs (ea4e6e4c-5851-40e2-99f6-31b4f34e1feb): starting stripes_read
[  746.159099] bcachefs (ea4e6e4c-5851-40e2-99f6-31b4f34e1feb): stripes_read done
[  746.159101] bcachefs (ea4e6e4c-5851-40e2-99f6-31b4f34e1feb): starting mark and sweep
[  746.249348] bcachefs (ea4e6e4c-5851-40e2-99f6-31b4f34e1feb): superblock not marked as containing replicas (type 14), fixing
[  746.289999] bcachefs (ea4e6e4c-5851-40e2-99f6-31b4f34e1feb): superblock not marked as containing replicas (type 14), fixing
[  746.366021] bcachefs (ea4e6e4c-5851-40e2-99f6-31b4f34e1feb): superblock not marked as containing replicas (type 14), fixing
[  746.450013] bcachefs (ea4e6e4c-5851-40e2-99f6-31b4f34e1feb): superblock not marked as containing replicas (type 14), fixing
[  782.934213] bcachefs (ea4e6e4c-5851-40e2-99f6-31b4f34e1feb): stripe 0 has wrong block_sectors[0]: got 16, should be 32, fixing
[  782.934216] bcachefs (ea4e6e4c-5851-40e2-99f6-31b4f34e1feb): stripe 0 has wrong block_sectors[1]: got 8, should be 16, fixing
[  782.934217] bcachefs (ea4e6e4c-5851-40e2-99f6-31b4f34e1feb): stripe 1 has wrong block_sectors[0]: got 584, should be 1168, fixing
[  782.934218] bcachefs (ea4e6e4c-5851-40e2-99f6-31b4f34e1feb): stripe 1 has wrong block_sectors[1]: got 64, should be 128, fixing
[  782.934219] bcachefs (ea4e6e4c-5851-40e2-99f6-31b4f34e1feb): stripe 3 has wrong block_sectors[0]: got 32, should be 64, fixing
[  782.934220] bcachefs (ea4e6e4c-5851-40e2-99f6-31b4f34e1feb): stripe 4 has wrong block_sectors[0]: got 16, should be 32, fixing
[  782.934221] bcachefs (ea4e6e4c-5851-40e2-99f6-31b4f34e1feb): stripe 4 has wrong block_sectors[1]: got 8, should be 16, fixing
[  782.934222] bcachefs (ea4e6e4c-5851-40e2-99f6-31b4f34e1feb): stripe 6 has wrong block_sectors[0]: got 96, should be 192, fixing
[  782.934223] bcachefs (ea4e6e4c-5851-40e2-99f6-31b4f34e1feb): stripe 6 has wrong block_sectors[1]: got 120, should be 240, fixing
[  782.934224] bcachefs (ea4e6e4c-5851-40e2-99f6-31b4f34e1feb): stripe 7 has wrong block_sectors[0]: got 200, should be 400, fixing
[  782.934225] bcachefs (ea4e6e4c-5851-40e2-99f6-31b4f34e1feb): Ratelimiting new instances of previous error
[  782.957536] bcachefs (ea4e6e4c-5851-40e2-99f6-31b4f34e1feb): bucket 1:374531 gen 9 data type user has wrong cached_sectors: got 440, should be 448, fixing
[  782.959483] bcachefs (ea4e6e4c-5851-40e2-99f6-31b4f34e1feb): bucket 1:870245 gen 20 data type user has wrong cached_sectors: got 88, should be 176, fixing
[  783.109272] bcachefs (ea4e6e4c-5851-40e2-99f6-31b4f34e1feb): fs has wrong user: 2/3 [0 3 5]: got 792096, should be 785784, fixing
[  783.109291] bcachefs (ea4e6e4c-5851-40e2-99f6-31b4f34e1feb): fs has wrong user: 2/3 [0 2 3]: got 1437840, should be 1420016, fixing
[  783.109296] bcachefs (ea4e6e4c-5851-40e2-99f6-31b4f34e1feb): fs has wrong user: 2/3 [0 1 2]: got 64, should be 0, fixing
[  783.109311] bcachefs (ea4e6e4c-5851-40e2-99f6-31b4f34e1feb): fs has wrong user: 2/3 [0 2 5]: got 820168, should be 823464, fixing
[  783.109312] bcachefs (ea4e6e4c-5851-40e2-99f6-31b4f34e1feb): fs has wrong user: 2/3 [2 3 5]: got 589755, should be 610659, fixing
[  783.113251] bcachefs (ea4e6e4c-5851-40e2-99f6-31b4f34e1feb): mark and sweep done
[  783.113252] bcachefs (ea4e6e4c-5851-40e2-99f6-31b4f34e1feb): starting journal replay
[  789.424908] bcachefs (ea4e6e4c-5851-40e2-99f6-31b4f34e1feb): journal replay done
[  789.424923] bcachefs (ea4e6e4c-5851-40e2-99f6-31b4f34e1feb): writing allocation info
[  789.424992] ------------[ cut here ]------------
[  789.424993] kernel BUG at fs/bcachefs/ec.c:586!
[  789.424999] invalid opcode: 0000 [#1] SMP PTI
[  789.425001] CPU: 1 PID: 8173 Comm: mount Tainted: P           O      5.10.0+bcachefs.git20210112.03b7eec4-1-generic #1-Ubuntu
[  789.425002] Hardware name: System manufacturer System Product Name/Z170 PRO GAMING, BIOS 1904 07/05/2016
[  789.425018] RIP: 0010:heap_verify_backpointer+0x84/0x90 [bcachefs]
[  789.425020] Code: 40 0d 01 74 23 48 8b 00 49 3b 84 24 60 f5 01 00 73 18 48 c1 e0 04 49 03 84 24 68 f5 01 00 48 39 18 75 09 5b 41 5c 5d c3 0f 0b <0f> 0b 0f 0b 0f 1f 84 00 00 00 00 00 0f 1f 44 00 00 55 31 c9 48 89
[  789.425021] RSP: 0018:ffffae14c2c67770 EFLAGS: 00010206
[  789.425023] RAX: 0000000000000005 RBX: 0000000000000000 RCX: 0000000000000003
[  789.425024] RDX: ffffffffffffffff RSI: 0000000000000000 RDI: ffff9f1d0ea0e000
[  789.425024] RBP: ffffae14c2c67780 R08: ffff9f1d0ea0d000 R09: 0000000000000015
[  789.425025] R10: 0000000000000001 R11: 0000000000000001 R12: ffff9f1cf7b00000
[  789.425026] R13: ffff9f1cf7b1f570 R14: ffff9f1c917f9028 R15: 0000000000000001
[  789.425027] FS:  00007ff344b33800(0000) GS:ffff9f22a6c40000(0000) knlGS:0000000000000000
[  789.425028] CS:  0010 DS: 0000 ES: 0000 CR0: 0000000080050033
[  789.425029] CR2: 000000c0001b9000 CR3: 000000026c036002 CR4: 00000000003706e0
[  789.425030] DR0: 0000000000000000 DR1: 0000000000000000 DR2: 0000000000000000
[  789.425030] DR3: 0000000000000000 DR6: 00000000fffe0ff0 DR7: 0000000000000400
[  789.425031] Call Trace:
[  789.425044]  bch2_stripes_heap_update+0x3d/0x370 [bcachefs]
[  789.425052]  ? bch2_btree_node_iter_init+0x63/0x840 [bcachefs]
[  789.425062]  bch2_mark_stripe+0x2e9/0x560 [bcachefs]
[  789.425072]  bch2_mark_key_locked+0x122/0x1f0 [bcachefs]
[  789.425081]  bch2_mark_update+0x163/0x470 [bcachefs]
[  789.425090]  bch2_trans_commit_write_locked+0x26b/0x580 [bcachefs]
[  789.425099]  do_bch2_trans_commit+0x29d/0x780 [bcachefs]
[  789.425108]  __bch2_trans_commit+0x756/0x860 [bcachefs]
[  789.425118]  ? __bch2_stripe_write_key+0x16e/0x1a0 [bcachefs]
[  789.425128]  bch2_stripes_write+0x1b5/0x220 [bcachefs]
[  789.425138]  ? bch2_stripes_write+0x6e/0x220 [bcachefs]
[  789.425150]  bch2_fs_recovery+0x494/0x8d0 [bcachefs]
[  789.425153]  ? __closure_wake_up+0x1a/0x250
[  789.425165]  bch2_fs_start+0x413/0x5c0 [bcachefs]
[  789.425176]  ? bch2_fs_start+0x413/0x5c0 [bcachefs]
[  789.425186]  bch2_fs_open+0x303/0x3b0 [bcachefs]
[  789.425197]  bch2_mount+0x3d4/0x530 [bcachefs]
[  789.425200]  legacy_get_tree+0x2b/0x50
[  789.425202]  vfs_get_tree+0x2a/0xc0
[  789.425203]  do_new_mount+0x14b/0x1a0
[  789.425204]  path_mount+0x1d4/0x4e0
[  789.425206]  __x64_sys_mount+0x108/0x140
[  789.425208]  do_syscall_64+0x38/0x90
[  789.425210]  entry_SYSCALL_64_after_hwframe+0x44/0xa9
[  789.425211] RIP: 0033:0x7ff344d8fd1e
[  789.425213] Code: 48 8b 0d 4d 81 0c 00 f7 d8 64 89 01 48 83 c8 ff c3 66 2e 0f 1f 84 00 00 00 00 00 90 f3 0f 1e fa 49 89 ca b8 a5 00 00 00 0f 05 <48> 3d 01 f0 ff ff 73 01 c3 48 8b 0d 1a 81 0c 00 f7 d8 64 89 01 48
[  789.425214] RSP: 002b:00007ffcb9290938 EFLAGS: 00000246 ORIG_RAX: 00000000000000a5
[  789.425215] RAX: ffffffffffffffda RBX: 00007ff344ebb264 RCX: 00007ff344d8fd1e
[  789.425216] RDX: 000055f3d7de5c60 RSI: 000055f3d7de7a30 RDI: 000055f3d7de5cc0
[  789.425217] RBP: 000055f3d7de5a30 R08: 000055f3d7de5c80 R09: 000055f3d7de7a50
[  789.425217] R10: 0000000000000000 R11: 0000000000000246 R12: 0000000000000000
[  789.425218] R13: 000055f3d7de5cc0 R14: 000055f3d7de5c60 R15: 000055f3d7de5a30
[  789.425219] Modules linked in: poly1305_generic libpoly1305 poly1305_x86_64 chacha_generic chacha_x86_64 libchacha bcachefs crc64 xor raid6_pq libcrc32c nfnetlink zfs(PO) zunicode(PO) zavl(PO) icp(PO) zcommon(PO) znvpair(PO) spl(O) zlua(PO) intel_rapl_msr intel_rapl_common x86_pkg_temp_thermal intel_powerclamp coretemp kvm_intel kvm snd_hda_codec_realtek snd_hda_codec_generic ledtrig_audio snd_hda_codec_hdmi snd_hda_intel snd_intel_dspcfg soundwire_intel soundwire_generic_allocation soundwire_cadence snd_hda_codec hid_multitouch mei_hdcp joydev input_leds snd_hda_core snd_hwdep soundwire_bus snd_soc_core snd_compress ac97_bus snd_pcm_dmaengine snd_pcm mei_me mei snd_seq_midi snd_seq_midi_event rapl intel_cstate snd_rawmidi snd_seq snd_seq_device snd_timer eeepc_wmi asus_wmi snd sparse_keymap serio_raw wmi_bmof efi_pstore soundcore ee1004 mac_hid acpi_pad sch_fq_codel msr parport_pc ppdev lp parport ip_tables x_tables autofs4 overlay nls_utf8 isofs nls_iso8859_1 dm_mirror dm_region_hash
[  789.425254]  dm_log amdgpu iommu_v2 gpu_sched hid_logitech_hidpp hid_logitech_dj hid_generic usbhid hid uas usb_storage nouveau radeon mxm_wmi i2c_algo_bit ttm crct10dif_pclmul crc32_pclmul drm_kms_helper ghash_clmulni_intel aesni_intel syscopyarea sysfillrect sysimgblt fb_sys_fops cec crypto_simd cryptd nvme glue_helper rc_core ahci e1000e i2c_i801 psmouse drm i2c_smbus nvme_core libahci xhci_pci xhci_pci_renesas video wmi
[  789.425272] ---[ end trace e401916c2147f03c ]---
```
Resolved line numbers:
```
bch2_stripes_heap_update+0x3d/0x370:
bch2_stripes_heap_update at /build/linux-CfRv9T/linux-5.10.0+bcachefs.git20210112.03b7eec4/fs/bcachefs/ec.c:636

bch2_btree_node_iter_init+0x63/0x840:
bch2_btree_node_iter_init at /build/linux-CfRv9T/linux-5.10.0+bcachefs.git20210112.03b7eec4/fs/bcachefs/bset.c:1472

bch2_mark_stripe+0x2e9/0x560:
pv_queued_spin_unlock at /build/linux-CfRv9T/linux-5.10.0+bcachefs.git20210112.03b7eec4/arch/x86/include/asm/paravirt.h:559
(inlined by) queued_spin_unlock at /build/linux-CfRv9T/linux-5.10.0+bcachefs.git20210112.03b7eec4/arch/x86/include/asm/qspinlock.h:56
(inlined by) do_raw_spin_unlock at /build/linux-CfRv9T/linux-5.10.0+bcachefs.git20210112.03b7eec4/include/linux/spinlock.h:212
(inlined by) __raw_spin_unlock at /build/linux-CfRv9T/linux-5.10.0+bcachefs.git20210112.03b7eec4/include/linux/spinlock_api_smp.h:151
(inlined by) spin_unlock at /build/linux-CfRv9T/linux-5.10.0+bcachefs.git20210112.03b7eec4/include/linux/spinlock.h:394
(inlined by) bch2_mark_stripe at /build/linux-CfRv9T/linux-5.10.0+bcachefs.git20210112.03b7eec4/fs/bcachefs/buckets.c:1290

bch2_mark_key_locked+0x122/0x1f0:
bch2_mark_key_locked at /build/linux-CfRv9T/linux-5.10.0+bcachefs.git20210112.03b7eec4/fs/bcachefs/buckets.c:1335

bch2_mark_update+0x163/0x470:
bch2_mark_update at /build/linux-CfRv9T/linux-5.10.0+bcachefs.git20210112.03b7eec4/fs/bcachefs/buckets.c:1414

bch2_trans_commit_write_locked+0x26b/0x580:
bch2_trans_commit_write_locked at /build/linux-CfRv9T/linux-5.10.0+bcachefs.git20210112.03b7eec4/fs/bcachefs/btree_update_leaf.c:465

do_bch2_trans_commit+0x29d/0x780:
do_bch2_trans_commit at /build/linux-CfRv9T/linux-5.10.0+bcachefs.git20210112.03b7eec4/fs/bcachefs/btree_update_leaf.c:546

__bch2_trans_commit+0x756/0x860:
__bch2_trans_commit at /build/linux-CfRv9T/linux-5.10.0+bcachefs.git20210112.03b7eec4/fs/bcachefs/btree_update_leaf.c:944

__bch2_stripe_write_key+0x16e/0x1a0:
__bch2_stripe_write_key at /build/linux-CfRv9T/linux-5.10.0+bcachefs.git20210112.03b7eec4/fs/bcachefs/ec.c:1486

bch2_stripes_write+0x1b5/0x220:
bch2_trans_commit at /build/linux-CfRv9T/linux-5.10.0+bcachefs.git20210112.03b7eec4/fs/bcachefs/btree_update.h:101 (discriminator 2)
(inlined by) bch2_stripes_write at /build/linux-CfRv9T/linux-5.10.0+bcachefs.git20210112.03b7eec4/fs/bcachefs/ec.c:1510 (discriminator 2)

bch2_stripes_write+0x6e/0x220:
bch2_trans_get_iter at /build/linux-CfRv9T/linux-5.10.0+bcachefs.git20210112.03b7eec4/fs/bcachefs/btree_iter.h:272 (discriminator 2)
(inlined by) bch2_stripes_write at /build/linux-CfRv9T/linux-5.10.0+bcachefs.git20210112.03b7eec4/fs/bcachefs/ec.c:1503 (discriminator 2)

bch2_fs_recovery+0x494/0x8d0:
bch2_fs_recovery at /build/linux-CfRv9T/linux-5.10.0+bcachefs.git20210112.03b7eec4/fs/bcachefs/recovery.c:1132 (discriminator 3)
```

**Comment by koverstreet** [comment]
Fixed by 03b7eec4ff
